lib/sisimai/bite/email/google.rb

```diff
@@ -0,0 +1,320 @@
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # Technical details of permanent failure:=20
+            # Google tried to deliver your message, but it was rejected by the recipient =
+            # domain. We recommend contacting the other email provider for further inform=
+            # ation about the cause of this error. The error that the other server return=
+            # ed was: 554 554 5.7.0 Header error (state 18).
+            #
+            # -- OR --
+            #
+            # Technical details of permanent failure:=20
+            # Google tried to deliver your message, but it was rejected by the server for=
+            # the recipient domain example.jp by mx.example.jp. [192.0.2.49].
+            #
+            # The error that the other server returned was:
+            # 550 5.1.1 <userunknown@example.jp>... User Unknown
+            #
+            v = dscontents[-1]
+
+            if cv = e.match(/\A[ \t]+([^ ]+[@][^ ]+)\z/)
+              # kijitora@example.jp: 550 5.2.2 <kijitora@example>... Mailbox Full
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+
+              addr0 = Sisimai::Address.s3s4(cv[1])
+              if Sisimai::RFC5322.is_emailaddress(addr0)
+                v['recipient'] = addr0
+                recipients += 1
+              end
+
+            else
+              v['diagnosis'] ||= ''
+              v['diagnosis']  += e + ' '
+            end
+          end
+        end
+
+        return nil if recipients.zero?
+        require 'sisimai/string'
+        require 'sisimai/smtp/status'
+
+        dscontents.map do |e|
+          e['agent']     = self.smtpagent
+          e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+
+          unless e['rhost']
+            # Get the value of remote host
+            if cv = e['diagnosis'].match(/[ \t]+by[ \t]+([^ ]+)[.][ \t]+\[(\d+[.]\d+[.]\d+[.]\d+)\][.]/)
+              # Google tried to deliver your message, but it was rejected by
+              # the server for the recipient domain example.jp by mx.example.jp. [192.0.2.153].
+              hostname = cv[1]
+              ipv4addr = cv[2]
+              if hostname =~ /[-0-9a-zA-Z]+[.][a-zA-Z]+\z/
+                # Maybe valid hostname
+                e['rhost'] = hostname.downcase
+              else
+                # Use IP address instead
+                e['rhost'] = ipv4addr
+              end
+            end
+          end
+
+          if cv = e['diagnosis'].match(/[(]state[ ](\d+)[)][.]/)
+            statecode0 = cv[1]
+          end
+          if StateTable[statecode0]
```

Comment:
__[rubocop]__
* Layout/CommentIndentation: Incorrect indentation detected (column 16 instead of 29).

<!-- Comment by [SideCI](https://sideci.com) -->